addons/payment/models/payment_provider.py

```diff
@@ -515,7 +515,7 @@ def _get_compatible_providers(
             )
 
         # Handle tokenization support requirements.
-        if force_tokenization or self._is_tokenization_required(**kwargs):
+        if force_tokenization or (self._is_tokenization_required(**kwargs) and self._hide_non_tokenize_provider(**kwargs)):
```

Comment:
The `force_tokenization` argument and the `_is_tokenization_required` are hooks that `sale_subscription` already uses to indicate whether providers disallowing tokenization should be filtered out. If possible, you should update the way these hooks are used rather than add a third one that does the same job as the other two.

Reply:
Hello @AntoineVDV :wave: ,

We will now allow non-tokenized providers to pay for subscriptions. Previously, we checked two conditions: if either `force_tokenization` or `_is_tokenization_required` was true, we filtered out the providers. However, `_is_tokenization_required` is always true for subscriptions, which caused the providers to be filtered out. Changing the value of `_is_tokenization_required` to false affects other processes, such as saving tokens during subscription quote payments and displaying the save tokenization input box on the payment form.

To address this, I created another hook with a default value of True so that it does not affect the current flow. For subscription orders, I set it to False to prevent filtering out non-tokenized providers.

Reply:
> Changing the value of _is_tokenization_required to false affects other processes, such as saving tokens during subscription quote payments and displaying the save tokenization input box on the payment form.

The way I understand it, you need more context in `_is_tokenization_required` to determine which flow you are in. This context could be part of the `env._context` variable or a kwarg passed to the methods calling `_is_tokenization_required`.

I think that adding yet another hook in `payment` for a very specific need in `sale_subscription` is a bad idea, as it will make understanding and maintenance of the payment engine more complicated.

Reply:
I have made some changes. I added a new parameter `show_non_tokenize_provider`, to the `kwargs` when calling `_get_compatible_providers` from the subscription. In the `_is_tokenization_required` method for subscription orders, it now returns `False` if `show_non_tokenize_provider` is present in `kwargs`; otherwise, it returns `True`. This ensures that when `_is_tokenization_required` is called from `_get_compatible_providers`, it will return `False`, so it won't affect other flows.